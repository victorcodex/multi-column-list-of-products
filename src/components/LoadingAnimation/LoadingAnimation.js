import styled, { keyframes } from 'styled-components';

const LoadingAnimationContainer = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    min-height: 100vh;
`;

const animationKeyFrames = boxShadowColor =>
  keyframes`
    0% {
      box-shadow: 9984px 0 0 -1px ${boxShadowColor}, 9999px 0 0 1px ${boxShadowColor}, 10014px 0 0 -1px ${boxShadowColor};
    }
    50% {
      box-shadow: 10014px 0 0 -1px ${boxShadowColor}, 9984px 0 0 -1px ${boxShadowColor}, 9999px 0 0 1px ${boxShadowColor};
    }
    100% {
      box-shadow: 9999px 0 0 1px ${boxShadowColor}, 10014px 0 0 -1px ${boxShadowColor}, 9984px 0 0 -1px ${boxShadowColor};
    }
  `;

const LoadingAnimation = styled.div`
  position: relative;
  left: -9999px;
  width: 1vh;
  height: 1vh;
  border-radius: 1vh;
  animation: ${animationKeyFrames("gray")}
    ${props => props.delay} infinite linear;
`;

export {
    LoadingAnimationContainer,
    LoadingAnimation
}