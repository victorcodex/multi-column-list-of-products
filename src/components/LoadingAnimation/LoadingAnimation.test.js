import React from 'react';
import { render } from '@testing-library/react';

import { LoadingAnimation } from './LoadingAnimation';

describe('LoadingAnimation', () => {
  it('should render loading animation', () => {
    const { asFragment } = render(<LoadingAnimation delay="1.5s" />);
    expect(asFragment()).toMatchSnapshot();
  });
});