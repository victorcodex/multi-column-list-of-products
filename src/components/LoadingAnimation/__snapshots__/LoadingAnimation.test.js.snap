// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAnimation should render loading animation 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT dNCuZL"
  />
</DocumentFragment>
`;
